staging/src/k8s.io/apiserver/pkg/endpoints/request/context_shard.go

```diff
@@ -0,0 +1,49 @@
+package request
+
+import (
+	"context"
+)
+
+type shardKey int
+
+const (
+	// shardKey is the context key for the request namespace.
+	shardContextKey shardKey = iota
+)
+
+type Shard struct {
+	// Name is the name of the cluster.
+	Name string
```

Comment:
Given that we (likely) have shard name requirements, and `*` will not be a valid shard name, is there any reason not just have `type Shard string`? Otherwise, what is the meaning of:

```go
Shard{
  Name: "*",
  Wildcard: false,
}
```

or

```go
Shard{
  Name: "sapphire",
  Wildcard: true,
}
```
?

Reply:
Right, those are the happy paths. My question was about the others, which are (I think) not possible. In e.g. https://github.com/kcp-dev/kcp/pull/1841 you have a `New('*")` value for wildcard... seems like this could also ?

Reply:
How is `Name:"", Wildcard: false` different from `New("")`?

Reply:
My argument: the `Wildcard` field as you have written it is logically equivalent in every case to `obj.Name == "*"`

Reply:
@stevekuznetsov I updated the type to `type Shard string`, now `Shard("*")` indicates wildcard and `Shard("")` indicates a request without a shard name specified.

Reply:
I hope it is less confusing now.